server/src/subscriptions.js

```diff
@@ -1,29 +1,32 @@
-const {
-  execute,
-  subscribe
-} = require('graphql')
 const { PubSub } = require('apollo-server');
-const { SubscriptionServer } = require('subscriptions-transport-ws')
+const mqtt = require('mqtt')
+const { MQTTPubSub } = require('@aerogear/graphql-mqtt-subscriptions')
 
-function subscriptionServer (keycloakService, httpServer, apolloServer) {
-    return new SubscriptionServer({
-      execute,
-      subscribe,
-      onConnect: async connectionParams => {
-        if(keycloakService) {
-          return await keycloakService.validateToken(connectionParams)
-        } else {
-          return true;
-        }
-      },
-      schema: apolloServer.schema
-    }, {
-      server: httpServer,
-      path: '/graphql'
-    });
+const host = process.env.MQTT_HOST || 'localhost'
+
+const mqttOptions = {
+  host: host,
+  servername: host, // needed to work in OpenShift. Lookup SNI.
+  username: process.env.MQTT_USERNAME || '',
+  password: process.env.MQTT_PASSWORD || '' ,
+  port: process.env.MQTT_PORT || '1883',
+  protocol: process.env.MQTT_PROTOCOL || 'mqtt',
+  rejectUnauthorized: false
 }
 
+const client = mqtt.connect(host, mqttOptions)
+
+console.log(`attempting to connect to messaging service ${host}`)
+
+client.on('connect', () => {
+  console.log('connected to messaging service')
+})
+
+client.on('error', (error) => {
+  console.log('error with mqtt connection')
```

Comment:
Would this be propagated back to the method that publishes the event. 
I mean if you like to connect and kill the server - does publishing on resolver fail or it will publish still but we will get a message here.